qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/customers/CustomerDatabaseQueries.groovy

```diff
@@ -195,71 +197,99 @@ class CustomerDatabaseQueries {
         if (customerExists(customer)) {
             throw new DatabaseQueryException("Customer is already in the database.")
         }
-        int customerId = createNewCustomer(customer)
-        storeAffiliation(customerId, customer.affiliations)
+        Connection connection = databaseSession.getConnection()
+        connection.setAutoCommit(false)
+
+        connection.withCloseable {it ->
+            try {
+                int customerId = createNewCustomer(it, customer)
+                storeAffiliation(it, customerId, customer.affiliations)
+                connection.commit()
+            } catch (Exception e) {
+                log.error(e.message)
+                log.error(e.stackTrace.join("\n"))
+                connection.rollback()
+                connection.close()
+                throw new DatabaseQueryException("Could not create customer.")
+            }
+
+        }
     }
 
     private boolean customerExists(Customer customer) {
-        return true
+        String query = "SELECT * FROM customer WHERE first_name = ? AND last_name = ? AND email = ?"
+        Connection connection = databaseSession.getConnection()
```

Comment:
why do the other methods obtain a connection but here the connection is created within the class?

Reply:
because the other methods are part of a transaction, and a connection should not be closed during an transaction. Otherwise, a transaction cannot be rolled back.

This query is stand-alone, that is why it can create and close the connection from the pool directly.

Reply:
Thanks for the explanation. I will approve this PR.